elfi/result.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+from collections import OrderedDict
+import inspect
+import sys
+import io
+
+import elfi.visualization as vis
+
+
+"""
+Implementations related to results and post-processing.
+"""
+
+
+class Result(object):
+    """Container for results from ABC methods. Allows intuitive syntax for plotting etc.
+
+    Parameters
+    ----------
+    samples_list : list of np.arrays
+    nodes : list of parameter nodes
+    """
+    def __init__(self, samples_list, nodes, **kwargs):
+        self.samples = OrderedDict()
+        for ii, n in enumerate(nodes):
+            self.samples[n.name] = samples_list[ii]
+        self.n_samples = len(list(self.samples.values())[0])
+        self.n_params = len(self.samples)
+
+        # get name of the ABC method
+        stack10 = inspect.stack()[1][0]
+        self.method = stack10.f_locals["self"].__class__.__name__
+
+        # TODO: remove once interface fixed
+        # make sure none of the given kwargs already exist
+        for k in kwargs.keys():
+            if k in self.__dir__():
+                raise KeyError("Conflicting key: ", k)
+
+        self.__dict__.update(kwargs)
```

Comment:
Referring to the earlier discussion this would be better as either explicitly stated member variable names, or then a separate dict, e.g. `self.meta` as Antti suggested.